bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -990,6 +1011,78 @@ void asyncRecoveryAddEntry(final byte[] data, final int offset, final int length
         doAsyncAddEntry(op);
     }
 
+    private boolean isWritesetWritable(DistributionSchedule.WriteSet writeSet,
+                                       long key, int allowedNonWritableCount) {
+        if (allowedNonWritableCount < 0) {
+            allowedNonWritableCount = 0;
+        }
+
+        final int sz = writeSet.size();
+        final int requiredWritable = sz - allowedNonWritableCount;
```

Comment:
Nice, you are flipping the logic here. Any reason why caller's themselves can't do it? A reader needs to send 1, and the writer needs to send AQ. isn't it? Maybe there is a good reason.

Reply:
I like to have 'requiredWritableCount' than a negative 'allowedNonWritableCount' as I mentioned in my about comment, which I think is easy to understand. @dlg99  any reason to pass the -ve (not writable) and flip to writable in this routine?

Reply:
@dlg99 I am not doing a -1, but I think caller itself specifying the requiredWriteCount will make the logic clean. But, that is up to you. I don't need to review this point again.